### ISSUE TITLE
Support GCP's Workload Identity for K8s

### ISSUE THREAD

**Issue by hoangpham95** [thelastpickle/cassandra-medusa#558]
My project doesn't allow service account key creation due to security reason, and it's recommended that I use workload identity to configure access from my K8s cluster to other GCP resources. Is there a plan to support [workload identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) with Medusa for K8s?

**Comment by Miles-Garnsey** [comment]
Hi @hoangpham95 , can you provide a bit more detail on how you'd like this to work? Do you want to use workload identity as a means to provide credentials for bucket access?

**Comment by clearclaw** [comment]
Yes, this would be for bucket access -- same as using an instance role in AWS.
